src/System.Private.ServiceModel/src/System/ServiceModel/Channels/TextMessageEncoder.cs

```diff
@@ -546,7 +546,9 @@ public override Task<ArraySegment<byte>> WriteMessageAsync(Message message, int
                     await message.WriteMessageAsync(xmlWriter);
                     xmlWriter.WriteEndDocument();
                 }
-                await xmlWriter.FlushAsync();
+
+                // The underlying type of xmlWriter, XmlUTF8TextWriter, has not implemented FlushAsync() yet.
+                xmlWriter.Flush();
```

Comment:
The API is in the contract, but it’s not implemented yet.

From: Sajay Antony [mailto:notifications@github.com]
Sent: Friday, July 24, 2015 10:18 AM
To: dotnet/wcf
Cc: Shin Mao
Subject: Re: [wcf] Support TransferMode.Streamed. (#209)


In src/System.Private.ServiceModel/src/System/ServiceModel/Channels/TextMessageEncoder.cs<https://github.com/dotnet/wcf/pull/209#discussion_r35444082>:

> @@ -546,7 +546,9 @@ public override Task<ArraySegment<byte>> WriteMessageAsync(Message message, int

>                      await message.WriteMessageAsync(xmlWriter);

>                      xmlWriter.WriteEndDocument();

>                  }

> -                await xmlWriter.FlushAsync();

> +

> +                // The underlying type of xmlWriter, XmlUTF8TextWriter, has not implemented FlushAsync() yet.

> +                xmlWriter.Flush();

The api is not exposed by the XmlWriter contract?

—
Reply to this email directly or view it on GitHub<https://github.com/dotnet/wcf/pull/209/files#r35444082>.